storage/connect/CMakeLists.txt

```diff
@@ -443,12 +440,17 @@ ENDIF()
 IF(CONNECT_WITH_JDBC AND JAVA_FOUND AND JNI_FOUND)
   # TODO: Find how to compile and install the java wrapper classes
   # Find required libraries and include directories
+  INCLUDE(UseJava)
   SET (JAVA_SOURCES JdbcInterface.java)
-  add_jar(JdbcInterface ${JAVA_SOURCES})
+  add_jar(JdbcInterface ${JAVA_SOURCES} OUTPUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/connect_jars)
+  get_target_property(_jarFile JdbcInterface JAR_FILE)
   INSTALL(FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/mysql-test/connect/std_data/JavaWrappers.jar
-    ${CMAKE_CURRENT_BINARY_DIR}/JdbcInterface.jar
-    DESTINATION ${INSTALL_MYSQLSHAREDIR} COMPONENT connect-engine)
+    ${_jarFile} DESTINATION ${INSTALL_MYSQLSHAREDIR} COMPONENT connect-engine-jdbc)
+  SET(CPACK_COMPONENTS_ALL ${CPACK_COMPONENTS_ALL} connect-engine-jdbc PARENT_SCOPE)
+  SET(CPACK_RPM_connect-engine-jdbc_PACKAGE_SUGGESTS "connect-engine libmariadb-java postgresql-jdbc mysql-connector-java" PARENT_SCOPE)
```

Comment:
`connect-engine`- what package is that? `MariaDB-connect-engine`

deb packages aren't names the same as rpm
`libmariadb-java` -> `mariadb-java-client`
https://packages.fedoraproject.org/pkgs/mariadb-java-client/mariadb-java-client/

Good to see postgresql-jdbc mysql-connector-java existing.

Reply:
`FIND_FILE(Libjni_h jni.h PATHS ${JAVA_INCLUDE_PATH})` hasn't been added to populate `CPACK_RPM_BUILDREQUIRES` (happens automagicly as described before, no manual statement needed, though do check on spec file/rpm produced).

and:

````
IF(NOT JAVA_JVM_LIBRARY_DEP)
  MESSAGE(FATAL_ERROR "No JAVA_JVM_LIBRARY_DEP set by FIND_JAVA)
ENDIF()
CPACK_RPM_connect-engine-jdbc_PACKAGE_REQUIRES(${JAVA_JVM_LIBRARY_DEP})
````
hasn't been set.

Reply:
> `connect-engine`- what package is that? `MariaDB-connect-engine`
> 
> deb packages aren't names the same as rpm `libmariadb-java` -> `mariadb-java-client` https://packages.fedoraproject.org/pkgs/mariadb-java-client/mariadb-java-client/
> 
> Good to see postgresql-jdbc mysql-connector-java existing.

:+1: . Recently have created [MDBF-555](https://jira.mariadb.org/browse/MDBF-555) to have overview of all packages and distros in KB, probably will do it soon, with PR #2654.


> `FIND_FILE(Libjni_h jni.h PATHS ${JAVA_INCLUDE_PATH})` hasn't been added to populate `CPACK_RPM_BUILDREQUIRES` (happens automagicly as described before, no manual statement needed, though do check on spec file/rpm produced).
> 
> and:
> 
> ```
> IF(NOT JAVA_JVM_LIBRARY_DEP)
>   MESSAGE(FATAL_ERROR "No JAVA_JVM_LIBRARY_DEP set by FIND_JAVA")
> ENDIF()
> CPACK_RPM_connect-engine-jdbc_PACKAGE_REQUIRES(${JAVA_JVM_LIBRARY_DEP})
> ```
> 
> hasn't been set.

In the [comment](https://github.com/MariaDB/server/pull/2645#discussion_r1209078570) I said I have rebased current PR on top of PR #2630 that uses `find_package` and it is approved without need of `find_file`, so I didn't use it here. Do we really need `find_file` if `find_package` is required? Cmake will raise the error on build.

Reply:
https://github.com/MariaDB/server/pull/2645#discussion_r1206153612 - `find_file` is purely for population of the `CPACK_RPM_BUILD_REQUIRES`. Without it, RPM don't have the build depends.